tools/github-actions/audit/src/main.ts

```diff
@@ -5,54 +5,148 @@ import {GitHubAdvisoryId, NPMAuditReportV1, Severity} from 'audit-types';
+}
+
+/**
+ * Format the response from yarn 3 npm audit in a common interface that will be used to build the
+ * report summary
+ *
+ * @param response
+ * @param severityThreshold
+ */
+function computeYarn3Report(response: string, severityThreshold: Severity): OtterAuditReport {
+  core.info('Computing Report for Yarn 3');
+  const reportJson = JSON.parse(response) as Audit;
+  core.debug(response);
+  const nbVulnerabilities = Object.values(reportJson.metadata.vulnerabilities as { [key: string]: number } || {}).reduce((acc, curr) => acc + curr, 0);
+  let highestSeverityFound: Severity | undefined;
+  for (let index = severities.length; index >= 0; index--) {
+    const severity: Severity = severities[index];
+    if (reportJson.metadata.vulnerabilities[severity] > 0) {
+      highestSeverityFound = severity;
+      break;
+    }
+  }
+  return Object.values(reportJson.advisories as Readonly<Record<GitHubAdvisoryId, Advisory>>)
+    .reduce<OtterAuditReport>((currentVulnerabilities, advisory: Advisory) => {
+      core.info(`${severities.indexOf(severityThreshold)} - ${severities.indexOf(advisory.severity)}`);
+      if (severities.indexOf(severityThreshold) <= severities.indexOf(advisory.severity)) {
+        currentVulnerabilities.errors.push({severity: advisory.severity, overview: advisory.overview, moduleName: advisory.module_name});
+      } else {
+        currentVulnerabilities.warnings.push({severity: advisory.severity, overview: advisory.overview, moduleName: advisory.module_name});
+      }
+      return currentVulnerabilities;
+    }, {errors: [], warnings: [], nbVulnerabilities, highestSeverityFound} as OtterAuditReport);
+}
+
 async function run(): Promise<void> {
-  /**
-   * Severities supported by yarn npm audit from the lowest to the highest criticality
-   */
-  const severities: Severity[] = ['info', 'low', 'moderate', 'high', 'critical'];
-  const colors = ['', 'green', 'yellow', 'orange', 'red'];
 
   try {
     const severityConfig = core.getInput('severity') as Severity;
     const allWorkspaces = core.getInput('allWorkspaces') === 'true';
     const recursive = core.getInput('recursive') === 'true';
     const environment = core.getInput('environment');
+    const versionOutput = await getExecOutput('yarn --version', [], {cwd: process.env.GITHUB_WORKSPACE});
+    const version = Number.parseInt(versionOutput.stdout.split('.')[0], 10);
     const command = `yarn npm audit --environment ${environment} ${allWorkspaces ? '--all ' : ''}${recursive ? '--recursive ' : ''}--json`;
 
-    const {stdout: report} = await getExecOutput(command, [], {cwd: process.env.GITHUB_WORKSPACE});
+    const {stdout: report, stderr: err} = await getExecOutput(command, [], {cwd: process.env.GITHUB_WORKSPACE, ignoreReturnCode: true});
+    core.warning(err);
     core.setOutput('reportJSON', report);
+    const reportData: OtterAuditReport = version >= 4 ? computeYarn4Report(report, severityConfig) : computeYarn3Report(report, severityConfig);
```

Comment:
maybe we could use `semver` satisfies function here
```suggestion
    const reportData: OtterAuditReport = satisfies(versionOutput.stdout, '>= 4') ? computeYarn4Report(report, severityConfig) : computeYarn3Report(report, severityConfig);
```

Reply:
Not found of adding yet another package for something as simple